widgetssdk/src/main/java/com/glia/widgets/core/chathead/ChatHeadService.java

```diff
@@ -75,12 +75,13 @@ private void initChatHeadView(
                             layoutParams.x = Float.valueOf(x).intValue();
                             layoutParams.y = Float.valueOf(y).intValue();
                             windowManager.updateViewLayout(chatHeadView, layoutParams);
+                            controller.onChatHeadPositionChanged(layoutParams.x, layoutParams.y);
                         },
                         (v) -> controller.onChatHeadClicked()
                 ));
     }
 
-    private WindowManager.LayoutParams getLayoutParams() {
+    private WindowManager.LayoutParams getLayoutParams(ServiceChatBubbleController controller) {
```

Comment:
`getLayoutParams` only cares about position so no point to inject the whole controller. improves readability.